base/applications/taskmgr/procpage.c

```diff
@@ -1103,116 +1095,48 @@ static DWORD GetProcessExecutablePathById(DWORD dwProcessId, LPWSTR lpExePath, D
 void ProcessPage_OnProperties(void)
 {
     DWORD dwProcessId;
-    WCHAR szPath[MAX_PATH];
-    LPWSTR pszPath = NULL;
-    LPWSTR pszExePath = NULL;
+    WCHAR szExePath[MAX_PATH];
     DWORD dwLength;
     SHELLEXECUTEINFOW info = { 0 };
 
     dwProcessId = GetSelectedProcessId();
-    dwLength = GetProcessExecutablePathById(dwProcessId, szPath, _countof(szPath));
+    dwLength = GetProcessExecutablePathById(dwProcessId, szExePath, _countof(szExePath));
 
-    if (dwLength == 0)
+    if (dwLength == 0 || dwLength > _countof(szExePath))
     {
         return;
     }
-    else if (dwLength > _countof(szPath))
-    {
-        pszPath = HeapAlloc(GetProcessHeap(), 0, dwLength * sizeof(WCHAR));
-
-        if (!pszPath)
-        {
-            return;
-        }
-
-        if (GetProcessExecutablePathById(dwProcessId, pszPath, dwLength) == 0)
-        {
-            goto Cleanup;
-        }
-
-        pszExePath = pszPath;
-    }
-    else
-    {
-        pszExePath = szPath;
-    }
 
     info.cbSize = sizeof(SHELLEXECUTEINFOW);
     info.fMask = SEE_MASK_INVOKEIDLIST;
     info.hwnd = NULL;
     info.lpVerb = L"properties";
-    info.lpFile = pszExePath;
+    info.lpFile = szExePath;
     info.lpParameters = L"";
     info.lpDirectory = NULL;
     info.nShow = SW_SHOW;
     info.hInstApp = NULL;
 
     ShellExecuteExW(&info);
-
-Cleanup:
-
-    if (pszPath)
-    {
-        HeapFree(GetProcessHeap(), 0, pszPath);
-    }
 }
 
 void ProcessPage_OnOpenFileLocation(void)
 {
     DWORD dwProcessId;
-    WCHAR szPath[MAX_PATH];
-    LPWSTR pszPath = NULL;
-    LPWSTR pszExePath = NULL;
-    LPWSTR pszCmdLine = NULL;
+    WCHAR szExePath[MAX_PATH];
+    WCHAR szCmdLine[MAX_PATH + 10];
     DWORD dwLength;
 
     dwProcessId = GetSelectedProcessId();
-    dwLength = GetProcessExecutablePathById(dwProcessId, szPath, _countof(szPath));
+    dwLength = GetProcessExecutablePathById(dwProcessId, szExePath, _countof(szExePath));
```

Comment:
It seems pointless to add long path support here because Explorer won't accept long paths anyway.

Reply:
That one place is not fixed yet does not mean we shouldn't aim to fix other places,
otherwise this would always be stuck in a 'but some other place is not fixed yet' loop.

Reply:
I understand that MAX_PATH limit is an annoying issue of the Windows platform and it can't be fixed without writing a new set of API functions that support long path so adding support for long path now will make the execution slower and it won't work anyway.

But if you think that using dynamic allocation now is better than rely on MAX_PATH until ROS supports long path, I'll change.